File: backend/seed.py
from sqlalchemy.orm import Session
from app.db.database import SessionLocal, engine, Base
from app.models import models
from app.core.security import get_password_hash

def seed_db():
    db = SessionLocal()
    try:
        # Create Tables
        Base.metadata.create_all(bind=engine)
        
        # Check if users exist
        if db.query(models.User).first():
            print("Database already seeded.")
            return

        # Users to seed
        users = [
            {
                "email": "admin@test.com",
                "password": "admin123",
                "role": "admin",
                "name": "Admin User"
            },
            {
                "email": "doctor@test.com",
                "password": "doctor123",
                "role": "doctor",
                "name": "Dr. Sarah"
            },
            {
                "email": "patient@test.com",
                "password": "patient123",
                "role": "patient",
                "name": "John Doe"
            }
        ]

        for u in users:
            user = models.User(
                email=u["email"],
                hashed_password=get_password_hash(u["password"]),
                name=u["name"],
                role=u["role"]
            )
            db.add(user)
        
        db.commit()
        print("Database seeded!")
    except Exception as e:
        print(f"Error seeding DB: {e}")
    finally:
        db.close()

if __name__ == "__main__":
    seed_db()

File: server/seed.ts
import sequelize from './config/db';
import User from './models/User';
import bcrypt from 'bcryptjs';

const seed = async () => {
    try {
        await sequelize.sync({ force: true }); // Reset DB

        const salt = await bcrypt.genSalt(10);
        const doctorPass = await bcrypt.hash('doctor123', salt);
        const patientPass = await bcrypt.hash('patient123', salt);
        const adminPass = await bcrypt.hash('admin123', salt);

        await User.bulkCreate([
            {
                email: 'doctor@test.com',
                password_hash: doctorPass,
                role: 'doctor',
                name: 'Dr. Sarah Johnson'
            },
            {
                email: 'doctor2@test.com',
                password_hash: doctorPass,
                role: 'doctor',
                name: 'Dr. Michael Chen'
            },
            {
                email: 'patient@test.com',
                password_hash: patientPass,
                role: 'patient',
                name: 'John Smith'
            },
            {
                email: 'patient2@test.com',
                password_hash: patientPass,
                role: 'patient',
                name: 'Emily Davis'
            },
            {
                email: 'admin@test.com',
                password_hash: adminPass,
                role: 'admin',
                name: 'Super Admin'
            }
        ]);

        console.log('Database seeded successfully!');
        process.exit();
    } catch (error) {
        console.error('Error seeding database:', error);
        process.exit(1);
    }
};

seed();
